src/adapter/src/catalog/builtin.rs

```diff
@@ -2371,6 +2371,67 @@ JOIN mz_catalog.mz_roles role_owner ON role_owner.id = class_objects.owner_id
 WHERE mz_schemas.database_id IS NULL OR d.name = pg_catalog.current_database()",
 };
 
+pub const PG_DEPEND: BuiltinView = BuiltinView {
+    name: "pg_depend",
+    schema: PG_CATALOG_SCHEMA,
+    sql: "CREATE VIEW pg_catalog.pg_depend AS WITH class_objects AS (
+    SELECT 
+        id,
+        CASE
+            WHEN type = 'table'  THEN 'pg_tables'::pg_catalog.regclass::pg_catalog.oid
+            WHEN type = 'source' THEN 'pg_tables'::pg_catalog.regclass::pg_catalog.oid
+            WHEN type = 'view'   THEN 'pg_views'::pg_catalog.regclass::pg_catalog.oid
+            WHEN type = 'materialized-view' THEN 'pg_matviews'::pg_catalog.regclass::pg_catalog.oid
+        END classid,
+        schema_id
+    FROM mz_catalog.mz_relations
+    UNION ALL
+    SELECT mz_indexes.id, mz_indexes.oid, schema_id
+    FROM mz_catalog.mz_indexes
+    JOIN mz_catalog.mz_relations ON mz_indexes.on_id = mz_relations.id
+),
+
+current_objects AS (
+    SELECT 
+        class_objects.id,
+        classid,
+        mz_objects.oid,
+        CASE
+            WHEN mz_catalog.mz_schemas.name = 'pg_catalog' THEN 'i' 
+            WHEN mz_catalog.mz_schemas.name = 'mz_catalog' THEN 'i' 
+            WHEN mz_catalog.mz_schemas.name = 'mz_internal' THEN 'i'
+            -- We mark sources as internal because they
+            -- are exposed via pg_catalog
+            -- as tables, but the system will error
+            -- if a user attempts DROP TABLE <source>.
+            WHEN (
+                SELECT count(*) FROM mz_catalog.mz_sources 
+                WHERE id = class_objects.id
+            ) > 0 THEN 'i'
+            ELSE 'n'
+        END deptype
+    FROM class_objects
+    JOIN mz_catalog.mz_objects ON mz_objects.id = class_objects.id
+    JOIN mz_catalog.mz_schemas ON mz_schemas.id = class_objects.schema_id
+    LEFT JOIN mz_catalog.mz_databases d ON d.id = mz_schemas.database_id
+    WHERE mz_schemas.database_id IS NULL OR d.name = pg_catalog.current_database()
```

Comment:
Oooh, this is tricky. As written, this will limit the output of `pg_depend` to dependency edges that are _wholly_ constrained in the current database. That's ... probably right? But dependency edges in Materialize can span databases:

```
CREATE VIEW db1.public.v1 AS SELECT 1;
CREATE VIEW db2.public.v2 AS SELECT * FROM db1.public.v1;
```

There's no perfect choice for what to show in `pg_depend`. If the current database is `db2`, if you show the dependency on `v1`, some tools might get confused because the referenced OID for `v1` won't exist in `pg_views`. But, similarly, if you _don't_ show the dependencies, some tools might get confused because `v2` appears to have no dependencies when clearly it does.

I think what you have is the better of the two options, but wanted to call it out.

Reply:
I'll add a comment in the query calling this out as well.

Reply:
Comment added.